### PR TITLE
group-for-unknown-events

### PR DESCRIPTION
**Pull request by VitalyMashanov**

### CODE REVIEW

File: src/helpers/event.ts [hunk 1]
```diff
@@ -125,24 +125,41 @@ export const convertEventActionToEventTreeNode = (event: EventAction): EventTree
...
-			nano: 0,
-			epochSecond: 0,
-		},
-		endTimestamp: {
-			nano: 0,
-			epochSecond: 0,
-		},
-		successful: false,
-	};
+	return eventId === 'unknown-root'
```
Comment (rzvdv) [inline src/helpers/event.ts:128]: i dont think this function can be called with 'unknown-root' id

File: src/stores/events/EventsDataStore.ts [hunk 2]
```diff
@@ -281,13 +282,22 @@ export default class EventsDataStore {
 				}
 				parentNodes.unshift(rootNode);
 			}
+			if (rootNode.parentId === 'unknown-root') {
```
Comment (rzvdv) [inline src/stores/events/EventsDataStore.ts:285]: if you created object `unknownRoot`, use `unknownRoot.eventId`

File: src/stores/events/EventsDataStore.ts [hunk 3]
```diff
@@ -281,13 +282,22 @@ export default class EventsDataStore {
...
+				if (!this.eventsCache.has('unknown-root')) {
```
Comment (rzvdv) [inline src/stores/events/EventsDataStore.ts:286]: I think we can delete this block actually
and move this block here
```
  if (rootNode.parentId === 'unknown-root') {
		parentNodes.unshift(unknownRoot);
	}
```

File: src/stores/events/EventsDataStore.ts [hunk 4]
```diff
@@ -296,6 +306,7 @@ export default class EventsDataStore {
 				if (
 					rootNode &&
 					(isRootEvent(rootNode) || rootNode.isUnknown) &&
```
Comment (rzvdv) [inline src/stores/events/EventsDataStore.ts:308]: now it gets a little bit simplier
we want to add to rootEventIds only if its isRootEvent and `!this.rootEventIds.includes(rootNode.eventId)`

File: src/stores/events/EventsDataStore.ts [hunk 5]
```diff
@@ -320,7 +331,11 @@ export default class EventsDataStore {
 
 				const { isUnknown, parentId, eventId } = event;
 
-				if ((isRootEvent(event) || isUnknown) && !rootNodes.includes(eventId)) {
+				if (
+					(isRootEvent(event) || isUnknown) &&
```
Comment (rzvdv) [inline src/stores/events/EventsDataStore.ts:335]: now it gets a little bit simplier we want to add to rootEventIds only if its isRootEvent and !rootNodes.includes(rootNode.eventId)

File: src/stores/events/EventsDataStore.ts [hunk 6]
```diff
@@ -281,23 +282,27 @@ export default class EventsDataStore {
 				}
 				parentNodes.unshift(rootNode);
 			}
+			if (rootNode.parentId === unknownRoot.eventId) {
+				if (!this.eventsCache.has(unknownRoot.eventId)) {
```
Comment (rzvdv) [inline src/stores/events/EventsDataStore.ts:286]: i think we can delete this inner if-block aswell

File: src/stores/events/EventsStore.ts [hunk 7]
```diff
@@ -253,7 +259,12 @@ export default class EventsStore {
 
 	@action
 	public selectNode = (eventTreeNode: EventTreeNode | null) => {
-		if (eventTreeNode === null || eventTreeNode.eventId !== this.selectedNode?.eventId) {
+		if (
+			eventTreeNode === null ||
+			(eventTreeNode.eventId !== this.selectedNode?.eventId &&
+				eventTreeNode.eventId !== unknownRoot.eventId &&
```
Comment (rzvdv) [inline src/stores/events/EventsStore.ts:265]: this function shouldn't be executed at all if node isUknown

**Comment by VekesserVladimir** [comment]
Reimplemented in https://github.com/th2-net/th2-rpt-viewer/pull/325

**Pull request closed without merging**